### DISCUSSION TITLE
WebGUI is not working properly behing ngnix proxy manager

### DISCUSSION THREAD

**Discussion by kurokawaikki** [firefly-iii/firefly-iii#7907] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I installed the FireflyIII via docker compose. Both of the env files are properly set.
I can visit and use the webGUI smoothly with http://my-ip-address:8080 without any issue.
However, when I use it behind the ngnix proxy manager, the webGUI is not working...
I can login and go to the dashboard. But all the money, graph will be gone.
Additionlly, the menu will also be broken. For example, if I clicked on the options, nothing happened...
Usually, after clicking the options, the menu will open showing the sub menu of profile, prefernces and etc. This problem applys to every clickable object. 
It seems like some function is not properly loaded. But I don't know how to fix this...

I confirmed every setp to make the FireflyIII to work behind reverse proxy.
I set the TRUSTED_PROXY to ** and APP_URL to my domain already.
But... still not working...
Can you help me to fix this issue?
Thank you very much.

### Debug information

No error message.
The problem is related to the webGUI page.
Server is working if I access the webGUI page via IP address directly with out ngnix proxy manager.

### Expected behaviour

WebGUI is working properly.
Every clickable objects and view objects are working.

### Steps to reproduce

1. Install ngnix proxy manager 
2. Intall FireFlyIII with docker compose
3. Set ngnix proxy manager as reverse proxy
4. Access the FireflyIII webGUI via reverse proxy (https://fireflyIII.mydomain.com)

**Comment by JC5** [comment]
Bad configuration on your end. I bet the browser console (ctrl-shift-i) gives more info.

**Reply by kurokawaikki** [reply]
I found two error from the console:
1. Refused to load the script 'https://fireflyIII.mydomain.com/cdn-cgi/scripts/7d0fa10a/cloudflare-static/rocket-loader.min.js' because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'strict-dynamic' 'self' 'unsafe-inline' 'nonce-zyWE1yy1XD0loizLdn0jaw=='". Note that 'strict-dynamic' is present, so host-based allowlisting is disabled. Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.
2. Refused to load the script 'https://fireflyIII.mydomain.com/cdn-cgi/scripts/5c5dd728/cloudflare-static/email-decode.min.js' because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'strict-dynamic' 'self' 'unsafe-inline' 'nonce-zyWE1yy1XD0loizLdn0jaw=='". Note that 'strict-dynamic' is present, so host-based allowlisting is disabled. Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.

**Reply by JC5** [reply]
Ah, you are hosting behind Cloudflare. Those scripts aren't mine but courtesy of Cloudflare. Firefly III naturally refuses to load such external scripts. However, that shouldn't stop the rest of the website from loading. Any other errors?

**Reply by kurokawaikki** [reply]
In the browser console there is no other error message.
I found another error in the login page.
DevTools failed to load source map: Could not load content for https://fireflyIII.mydomain.com/v3-local/dist/css/adminlte.min.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

I could not figure out... Because my other apps work fine behind npm with the similar reverse proxy setting.
May be some incompatibility between npm and fireflyIII?

**Reply by kurokawaikki** [reply]
I deploy again without npm but nginx again (with the fresh install of every thing, I can assure my configuration is good) with the nginx setting that is the same as posted here:
https://github.com/orgs/firefly-iii/discussions/5175
However, the problem is the same...
Could you give me some hint to solve this? Thank you very much.

**Reply by JC5** [reply]
If there are no errors in the browser console I have nothing to go on.

**Reply by kurokawaikki** [reply]
OK...
I gave up. Anyway thank you very much.
I will go for Actualbudget. :)

**Reply by JC5** [reply]
Great! I'm still curious to learn why it doesn't work, even though it seems there are no errors in the browser console or anywhere else. If you ever find out, let me know 👍

**Reply by BogTheMudWing** [reply]
This issue appears to occur when you use SSL. The HTML calls for resources using `http://example.com/build/assets`. When I disabled *Force SSL* in Nginx Proxy Manager, the styling and scripts worked as intended on http. Trying to load the page with https still causes everything to break.

**Reply by BogTheMudWing** [reply]
I think this is because the references use absolute locations rather than relative. Firefly III assumes the client is connecting via HTTP, but browsers do not allow mixed HTTP and HTTPS, and so they are simply not loaded when using HTTPS.

**Reply by BogTheMudWing** [reply]
Ah, and after some more searching, I've found this is actually mentioned in the documentation.

> All links refer to http so forms and charts don't work
>
> This usually happens when you did not set TRUSTED_PROXIES=*.

Everything now works correctly! Thank you!
